tests/test_ecc_circuit.py

```diff
@@ -0,0 +1,93 @@
+import pytest
+from zkevm_specs.ecc_circuit import (
+    EcAdd,
+    verify_circuit,
+    EccCircuit,
+)
+
+
+def verify(
+    circuit: EccCircuit,
+    success: bool = True,
+):
+    """
+    Verify the circuit with the assigned witness.
+    If `success` is False, expect the verification to fail.
+    """
+
+    exception = None
+    try:
+        verify_circuit(circuit)
+    except Exception as e:
+        exception = e
+    if success:
+        if exception:
+            raise exception
+        assert exception is None
+    else:
+        assert exception is not None
+
+
+def gen_ecAdd_testing_data():
+    normal = (
+        EcAdd(
+            p=(1, 2),
+            q=(1, 2),
+            out=(
+                0x030644E72E131A029B85045B68181585D97816A916871CA8D3C208C16D87CFD3,
+                0x15ED738C0E0A7C92E7845F96B2AE9C0A68A6A449E3538FC7FF3EBF7A5A18A2C4,
+            ),
+        ),
+        True,
+    )
+    # p is not on the curve
+    invalid_p = (
+        EcAdd(
+            p=(2, 3),
+            q=(1, 2),
+            out=(0, 0),
+        ),
+        True,
+    )
+    incorrect_out = (
+        EcAdd(
+            p=(1, 2),
+            q=(1, 2),
+            out=(
+                0x030644E72E131A029B85045B68181585D97816A916871CA8D3C208C16D87CFD0,
+                0x15ED738C0E0A7C92E7845F96B2AE9C0A68A6A449E3538FC7FF3EBF7A5A18A2C4,
+            ),
+        ),
+        False,
+    )
+    # q = -p
+    # py_ecc doesn't support this case, it returns (0x30644E72E131A029B85045B68181585D97816A916871CA8D3C208C16D87CFD45, 1)
+    # p_plus_neg_p = (
```

Comment:
`py_ecc` should support this case, it represents point at infinity by `None`, see https://github.com/ethereum/py_ecc/blob/master/py_ecc/bn128/bn128_curve.py#L92

Reply:
fixed in [c1d5840](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/500/commits/c1d5840498ec3fe3bc0a29d516f0944399dadcb9)